pkg/cli/startbuild/startbuild.go

```diff
@@ -513,6 +526,112 @@ func (o *StartBuildOptions) streamBuildLogs(build *buildv1.Build) error {
 	return err
 }
 
+// checkNonExistantResources checks for Resources (config maps, secrets, etc) listed in the BuildConfig
+// and returns error in case any of them are missing, or in case of unexpected behavior
+func (o *StartBuildOptions) checkNonExistantResources() error {
+	buildConfig, err := o.BuildClient.BuildConfigs(o.Namespace).Get(context.TODO(), o.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	cms := make([]string, 0, 3)
+	secrets := make([]string, 0, 3)
+
+	// compile lists of all ConfigMap and Secret names referred to by the BuildConfig
+	for _, cm := range buildConfig.Spec.Source.ConfigMaps {
+		cms = append(cms, cm.ConfigMap.Name)
+	}
+	for _, secret := range buildConfig.Spec.Source.Secrets {
+		secrets = append(secrets, secret.Secret.Name)
+	}
+
+	var volumes []buildv1.BuildVolume
+	switch buildConfig.Spec.Strategy.Type {
+	case buildv1.DockerBuildStrategyType:
+		volumes = buildConfig.Spec.Strategy.DockerStrategy.Volumes
+	case buildv1.SourceBuildStrategyType:
+		volumes = buildConfig.Spec.Strategy.SourceStrategy.Volumes
+	}
+
+	// add BuildVolume resources to the names lists
+	for _, v := range volumes {
+		switch v.Source.Type {
+		case buildv1.BuildVolumeSourceTypeSecret:
+			secrets = append(secrets, v.Source.Secret.SecretName)
+		case buildv1.BuildVolumeSourceTypeConfigMap:
+			cms = append(cms, v.Source.ConfigMap.Name)
+		}
+	}
+
+	// check for all the CM and Secret lists in order to find those
+	// that do not exist and get lists of those
+	missingCms, err := o.checkNonExistantConfigMaps(cms)
+	if err != nil {
+		return err
+	}
+	missingSecrets, err := o.checkNonExistantSecrets(secrets)
+	if err != nil {
+		return err
+	}
+
+	// join all the missing resources names together to form an error message
+	resourceMessages := make([]string, 0, 2)
+	if len(missingCms) > 0 {
+		resourceMessages = append(resourceMessages, fmt.Sprintf("Config Maps: %s", strings.Join(missingCms, ", ")))
+	}
+	if len(missingSecrets) > 0 {
+		resourceMessages = append(resourceMessages, fmt.Sprintf("Secrets: %s", strings.Join(missingSecrets, ", ")))
+	}
+
+	if len(resourceMessages) > 0 {
+		return fmt.Errorf("Missing resources\n%s", strings.Join(resourceMessages, "\n"))
+	}
+
+	return nil
+}
+
+// checkNonExistantConfigMaps takes a list of ConfigMap names and returns
+// list of all of those that do not exist in the namespace
+// or return error in case of unexpected behavior
+func (o *StartBuildOptions) checkNonExistantConfigMaps(cms []string) ([]string, error) {
+	result := make([]string, 0, 3)
```

Comment:
```suggestion
        if len(cms) == 0 {
            return []string{}, nil
        }
	result := make([]string, 0, 3)
```

Return quickly if there are no configMaps

Reply:
I'm not in favor of setting the capacity:
```suggestion
	result := []string{}
```